File: lib-database-entities/src/main/java/com/ustadmobile/lib/db/entities/DownloadJobItemWithDownloadSetItem.java
package com.ustadmobile.lib.db.entities;

import android.arch.persistence.room.Embedded;

import com.ustadmobile.lib.database.annotation.UmEmbedded;

/**
 * Combined DownloadJobItem with it's related DownloadSetItem - useful when running a download
 */
public class DownloadJobItemWithDownloadSetItem extends DownloadJobItem {

    public DownloadJobItemWithDownloadSetItem() {

    }

    @UmEmbedded
    @Embedded
    private DownloadSetItem downloadSetItem;

    public DownloadSetItem getDownloadSetItem() {
        return downloadSetItem;
    }

    public void setDownloadSetItem(DownloadSetItem downloadSetItem) {
        this.downloadSetItem = downloadSetItem;
    }
}

File: lib-database-entities/src/main/java/com/ustadmobile/lib/db/entities/DownloadJobItemWithDownloadSetItem.kt
package com.ustadmobile.lib.db.entities

import android.arch.persistence.room.Embedded
import com.ustadmobile.lib.database.annotation.UmEmbedded

/**
 * Combined DownloadJobItem with it's related DownloadSetItem - useful when running a download
 */
class DownloadJobItemWithDownloadSetItem : DownloadJobItem() {

    @UmEmbedded
    @Embedded
    var downloadSetItem: DownloadSetItem? = null
}
